src/Operators/vector_rotation_operators.jl

```diff
@@ -0,0 +1,157 @@
+# TODO: have a general Oceananigans-wide function that retrieves a pointwise
+# value for a function, an array, a number, a field etc?
+# This would be a generalization of `getbc` that could be used everywhere we need it
+@inline getvalue(::Nothing,        i, j, k, grid, args...) = nothing
+@inline getvalue(a::Number,        i, j, k, grid, args...) = a
+@inline getvalue(a::AbstractArray, i, j, k, grid, args...) = @inbounds a[i, j, k]
+
+"""
+    intrinsic_vector(i, j, k, grid::AbstractGrid, uₑ, vₑ, wₑ)
+
+Convert the three-dimensional vector with components `uₑ, vₑ, wₑ` defined in an _extrinsic_ 
+reference frame associated with the domain, to the reference frame _intrinsic_ to the grid. 
+
+_extrinsic_ reference frames are:
+
+- Cartesian for box domains 
+- Latitude - Longitude for spherical domains
+
+Therefore, for the [`RectilinearGrid`](@ref) and the [`LatitudeLongitudeGrid`](@ref), the _extrinsic_ and the 
+_intrinsic_ reference frames are equivalent. However, for other grids (e.g., for the
+ [`ConformalCubedSphereGrid`](@ref)) that might not be the case.
+"""
+@inline intrinsic_vector(i, j, k, grid::AbstractGrid, uₑ, vₑ, wₑ) = 
+    getvalue(uₑ, i, j, k, grid), getvalue(vₑ, i, j, k, grid), getvalue(wₑ, i, j, k, grid)
+
+"""
+    extrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ, wᵢ)
+
+Convert the three-dimensional vector with components `uᵢ, vᵢ, wᵢ ` defined on the _intrinsic_ reference 
+frame of the grid, to the _extrinsic_ reference frame associated with the domain.
+
+_extrinsic_ reference frames are:
+
+- Cartesian for box domains 
+- Latitude - Longitude for spherical domains
+
+Therefore, for the [`RectilinearGrid`](@ref) and the [`LatitudeLongitudeGrid`](@ref), the _extrinsic_ and the 
+_intrinsic_ reference frames are equivalent. However, for other grids (e.g., for the
+ [`ConformalCubedSphereGrid`](@ref)) that might not be the case.
+"""
+@inline extrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ, wᵢ) =
+    getvalue(uᵢ, i, j, k, grid), getvalue(vᵢ, i, j, k, grid), getvalue(wᵢ, i, j, k, grid)
+
+# 2D vectors
+@inline intrinsic_vector(i, j, k, grid::AbstractGrid, uₑ, vₑ) = 
+    getvalue(uₑ, i, j, k, grid), getvalue(vₑ, i, j, k, grid)
+
+@inline entrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ) = 
```

Comment:
```suggestion
@inline extrinsic_vector(i, j, k, grid::AbstractGrid, uᵢ, vᵢ) = 
```

or is it `intrinsic`? Is this tested?